cmd/notation/list.go

```diff
@@ -50,23 +51,41 @@ func runList(ctx context.Context, opts *listOpts) error {
 
 	// initialize
 	reference := opts.reference
+	if opts.localContent {
+		var layout ociLayout
+		var err error
+		layout.path, layout.reference, err = parseOCILayoutReference(opts.reference)
+		if err != nil {
+			return err
+		}
+		sigRepo, err := ociLayoutRepository(layout.path)
+		if err != nil {
+			return err
+		}
+		targetDesc, err := sigRepo.Resolve(ctx, layout.reference)
```

Comment:
Note: In general, `Resolve()` takes a tag or a digest and should not take a full reference while `Repository.Resolve()` can take any form.